packages/phoenix-evals/src/phoenix/evals/preview/tracing.py

```diff
@@ -0,0 +1,265 @@
+import logging
+from functools import wraps
+from inspect import BoundArguments, iscoroutinefunction, signature
+from typing import Any, Awaitable, Callable, Mapping, Optional, Sequence, TypeVar, cast, overload
+
+from openinference.semconv.trace import OpenInferenceSpanKindValues, SpanAttributes
+from opentelemetry import trace as trace_api
+from opentelemetry.trace import NoOpTracer, Status, StatusCode, Tracer
+from opentelemetry.util.types import AttributeValue
+from typing_extensions import ParamSpec
+
+logger = logging.getLogger(__name__)
+
+
+def get_tracer(tracer: Optional[Tracer] = None) -> Tracer:
+    """
+    1. Use the provided Tracer if given
+    2. Otherwise, pull from the global tracer provider
+    3. Fall back to NoOpTracer if all else fails
+    Args:
+        tracer_provider: Optional tracer provider to use. If None, will use global provider.
+    Returns:
+        A tracer instance
+    """
+
+    try:
+        if tracer is not None:
+            return tracer
+
+        global_tracer_provider = trace_api.get_tracer_provider()
+        return global_tracer_provider.get_tracer(__name__)
```

Comment:
ultimately the tracing here will probably have to conform to the OITracer in it's ability to redact certain attributes.